### PR TITLE
[WIP] Add definition handler.

### PR DESCRIPTION
**Pull request by ddunkin**
Add a definition handler for custom types. This allows the editor to jump to the definition of a type from a field declaration, for example.

### CODE REVIEW

**Comment by ddunkin** [comment]
@ejball this works, but it's kind of ugly. It would be nice if there was a better way to get from a position to an element in the service definition.

File: src/Facility.LanguageServer/Facility.LanguageServer.csproj [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    <LangVersion>Latest</LangVersion>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <PackageReference Include="Facility.Definition" Version="1.2.0" />
-    <PackageReference Include="OmniSharp.Extensions.LanguageServer" Version="0.7.7" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.0.0" />
-  </ItemGroup>
-</Project>
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (ejball) [inline src/Facility.LanguageServer/Facility.LanguageServer.csproj]: Newline change? I've taken to using `* text=auto eol=lf` in `.gitattributes` (see other Facility projects).

**Comment by ejball** [comment]
I don't know if my implementation would be much different than yours, but it still would be nice to support it at that level, I agree.

File: src/Facility.LanguageServer/Facility.LanguageServer.csproj (hunk 1 above)
Comment (ddunkin) [reply src/Facility.LanguageServer/Facility.LanguageServer.csproj]: I think I copied `.gitattributes` and `.editorconfig` from FacilityCSharp, which doesn't have the line ending settings.

**Pull request closed without merging**